hutch_python/epics_arch.py

```diff
@@ -0,0 +1,246 @@
+"""Module to help create the epicsArch file that will be read by the DAQ."""
+import argparse
+import logging
+import os
+import sys
+
+import happi
+from happi.backends.qs_db import QSBackend
+
+import hutch_python.constants
+
+from .constants import EPICS_ARCH_FILE_PATH
+
+logger = logging.getLogger(__name__)
+
+# Simple Argument Parser Setup
+parser = argparse.ArgumentParser(description='Create an epicsArch file from'
```

Comment:
I'd recommend refactoring to use the following structure:

```python
def _create_parser():
    ...
    return argparse.ArgumentParser()

def create_arch_file(experiment, hutch=None, path=None, dry_run=False):
    ...   # actually do the stuff here

def main():
    parser = _create_parser()
    parsed_args = parser.parse_args()
    kwargs = vars(parsed_args)
    create_arch_file(**kwargs)

if __name__ == '__main__':
   main()
```

1. It gives a clear `main()` entry point
2. The parser isn't global but created on-demand (and nicely tucked away in a function)
3. You can call `create_arch_file` from Python without making a list of string args
4. `main()` is small, only binding together argument parsing with calling the function that actually implements things
5. Overriding globals to implement the functionality as currently proposed is not a good thing (perhaps an anti-pattern even). Use default arguments for the function, and default arguments in the argument parser, and you sidestep these issues.

There are other things you could do to improve on what I'm suggesting, but I think it presents a reasonably good basis for a general CLI tool.

Reply:
Ok thanks for the feedback @klauer 
One question though:
- the `experiment` should be a required positional argument correct?, the user will not be able to do this:
change the path first:
`epicsarch-qs --path /change/the/path`
then create the file:
`epicsarch-qs experiment`

?

Reply:
You actually answered this already in `def create_arch_file(experiment, hutch=None, path=None, dry_run=False):` - but want to double check that this is what you meant

Reply:
Thanks for explaining this!

Reply:
No problem - hope you see the benefit in the above method. It may seem pretty simple, but I've only arrived at it after making many other badly-structured tools...

Reply:
Yeah it makes sense, you explained that to me in a different review a while ago, and I thought this time I actually got it, because I made the `create_file` "independent" - (or so I thought hahah) - but even the way I had it before it did not have all the parameters that the command line would - would not be the same.